File: Insertion_Sort/InsertionSort.js
function insertionSort(Arr){
	for (var i = 0; i < Arr.length; i++) {
		let key = Arr[i];
		let j = i;
		while(j>=1 && Arr[j-1]>key){
			Arr[j] = Arr[j-1];
			j--;
		}
		Arr[j] = key;
	}
}

function printArray(Arr){
	let output = "[";
	for (var i = 0; i < Arr.length; i++) {
		output+= " "+Arr[i]+" ";
		if (i<Arr.length-1) {output+=",";}
	}
	output += "]";

	console.log(output);
}

let Arr = [6,3,1,8,3,0,5];
insertionSort(Arr);
printArray(Arr);

File: Insertion_Sort/InsertionSort.R
N = as.integer(readline())
Arr = c()

for (i in 1:N) {
  Arr[i] = as.integer(readline())
}

insertionSort<-function(Arr){
  for (i in 2:length(Arr)) {
    key = Arr[i]
    j = i
    while(j>=2 && Arr[j-1]>key){
      Arr[j] = Arr[j-1]
      j = j - 1
    }
    Arr[j] = key
    
  }
  
  return(Arr);
}

printArray<-function(Arr){
  output = "["
  for (i in 1:length(Arr)) {
    output = paste(output," ",Arr[i]," ")
    if(i<length(Arr)){output = paste(output,",")}
  }
  output = paste(output,"]")
  print(output)
}

Arr = insertionSort(Arr)
printArray(Arr)


